pkg/apiserver/apiserver.go

```diff
@@ -130,7 +130,6 @@ func (c completedConfig) New() (*AdmissionServer, error) {
 		// TODO we're going to need a later k8s.io/apiserver so that we can get discovery to list a different group version for
 		// our endpoint which we'll use to back some custom storage which will consume the AdmissionReview type and give back the correct response
 		apiGroupInfo := genericapiserver.APIGroupInfo{
-			PrioritizedVersions :[]schema.GroupVersion{admissionv1beta1.SchemeGroupVersion},
```

Comment:
what was the logic to have the admission group here in the first place? Just looks wrong, doesn't it?

Reply:
Yes, it looks wrong to me. The only groups should come from the `AdmissionWebhook.Resource()` method.

Reply:
and funny that gofmt didn't complain. Looks like we don't run that in travis.

Reply:
This does not look like a gofmt issue. The `admission/v1beta1` is added as a prioritized group for any webhook. So, when I am using multiple groups, go-restful throws an error.

- https://github.com/kubernetes/apiserver/blob/release-1.11/pkg/server/genericapiserver.go#L328
- https://github.com/kubernetes/apiserver/blob/release-1.11/pkg/endpoints/groupversion.go#L100

Reply:
I meant the `<space>:`

Reply:
ah! :)

Reply:
Maybe the goal was to make this work even with empty version map. In that case we could skip the InstallAPIGroup call though.

Looking through the use of `PrioritizedVersions` in the generic api server, I see two loops, both continueing when there is no resource in that group:

- https://github.com/kubernetes/kubernetes/blob/cc5c17e554a4d8f802043b337ca0787ec0ce7475/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go#L378
- https://github.com/kubernetes/kubernetes/blob/cc5c17e554a4d8f802043b337ca0787ec0ce7475/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go#L323

Have you added a `debug.PrintTrace` for your error message why that has problem with this group in `PrioritizedVersions`?

Reply:
> Maybe the goal was to make this work even with empty version map. In that case we could skip the InstallAPIGroup call though.

Yeah, though skipping would be a fine substitute.

Reply:
Hi, I debugged the issue again. So, the error is coming from the discovery handler registry code. For this line : https://github.com/kubernetes/apiserver/blob/release-1.11/pkg/server/genericapiserver.go#L396

![webhook-error](https://user-images.githubusercontent.com/94814/42452653-8711d18e-833f-11e8-8933-d3347775e9c9.png)